### PR TITLE
Refactoring: Migrate to Poetry dependency management, packaging and PyPI distribution

### PR DESCRIPTION
**Pull request by priitlatt**
**Motivation**

The main goal of this pull request is to migrate the project to unified [PEP 518](https://peps.python.org/pep-0518/) compliant build system by using [Poetry](https://python-poetry.org/) dependency management and packaging tool.

**Current situation**

Up until now
- project dependencies were managed by [Pipenv](https://pipenv.pypa.io/),
- builds were run using [setuptools](https://github.com/pypa/setuptools) and
- PyPI distribution was done using [Twine](https://twine.readthedocs.io/en/stable/).

**What was done**

Old project configuration files `setup.cfg`, `Pipenv` and `setup.py` were replaced by `pyproject.toml` and `.flake8`.

Testing, building and distribution workflows in CI scripts were updated to use Poetry. New `release-test` CI workflow was added to publish test releases to [PyPI test environment](https://test.pypi.org/) which adds options to test distribution related changes.

Single code change renaming Ruby script `bin/code_signing_manager.rb` to `src/codemagic/scripts/code_signing_manager.rb`, accompanied by minor edit to [`code_signing_settings_manager.py`](https://github.com/codemagic-ci-cd/cli-tools/pull/244/files#diff-29c7306bd8a249866838c4c6f6eecdac2a8f9a7380c43783a5eb579a652bc87b), was required by the fact that there is no easy way to define non-Python shell scripts using Poetry.

[pre-commit](https://pre-commit.com/) hooks were introduced to ensure unified code style from all contributors without running unnecessary CI jobs.

Examples in readme file were brough up to date and information on how to set up development environment was added.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,52 +42,158 @@ to see general description and subcommands for the tool, or
...
+    And one of either
+    ```shell
+    export TEST_GCLOUD_SERVICE_ACCOUNT_CREDENTIALS_PATH=... # Path to gcloud service account creedentials with `JSON` key type
+    export TEST_GCLOUD_SERVICE_ACCOUNT_CREDENTIALS_CONTENT=... # Content of gcloud service account creedentials with `JSON` key type
+    ```
+
+### Pre-commit hooks
+
+Optionally [pre-commit](https://pre-commit.com/) framework can be used to ensure that
+source code updates are compliant with all the rules mentioned above.
```
Comment (helinanever) [inline README.md:190]:
```suggestion
the source code updates are compliant with all the rules mentioned above.
```

File: README.md [hunk 2]
```diff
@@ -35,52 +42,158 @@ to see general description and subcommands for the tool, or
...
+
+Installation instructions are available from their [docs](https://pre-commit.com/#installation).
```
Comment (helinanever) [inline README.md:192]:
```suggestion
Installation instructions are available in their [docs](https://pre-commit.com/#installation).
```

File: README.md [hunk 3]
```diff
@@ -35,52 +42,158 @@ to see general description and subcommands for the tool, or
...
+
+### Code style and formatting
+
+Automatic formatting checks are enforced using [Flake8](https://flake8.pycqa.org/en/latest/)
+and [isort](https://pycqa.github.io/isort/).
+
+Invoke Flake8 checks from repository root directory with
+
+```shell
+poetry run flake8 .
```
[the remark this answers is not in the document]
Comment (priitlatt) [reply README.md:127]: Yes, `.` is a reference to current directory.

File: README.md [hunk 4]
```diff
@@ -35,52 +42,158 @@ to see general description and subcommands for the tool, or
...
+```
+
+Ensure that all imports are ordered as expected using isort from repository root with
+
+```shell
+poetry run isort --check-only .
```
[the remark this answers is not in the document]
Comment (priitlatt) [reply README.md:133]: Same as above, it marks current directory and is an argument to `isort` command.

File: README.md [hunk 5]
```diff
@@ -1,25 +1,32 @@
 # Codemagic CLI Tools
 
-Command line utilities for managing mobile app builds, code signing, and deployment. These power mobile app builds at [codemagic.io](https://codemagic.io).
+Codemagic CLI Tools are a set of command line utilities for managing mobile app builds,
```
[the remark this answers is not in the document]
Comment (priitlatt) [reply README.md:3]: Duplicate of [another suggestion](https://github.com/codemagic-ci-cd/cli-tools/pull/244/files/cb7f414e5c9dd81e5c161c569564da4bf98626f5#r909610875).

**Review by remarkov — APPROVED** [review]

**Pull request merged**